Willi.PlateUpEnhancementMod/EventHandlers/NoClipHandler.cs

```diff
@@ -1,24 +1,37 @@
-﻿using System.Collections.Generic;
+﻿using Kitchen;
+using System.Collections.Generic;
 using UnityEngine;
 using Willi.PlateUpEnhancementMod.Helpers;
+using static Willi.PlateUpEnhancementMod.Config.ConfigHelper;
+
 
 namespace Willi.PlateUpEnhancementMod.EventHandlers
 {
     public static class NoClipHandler
     {
-        public static void SetNoClipState(bool isNoClip)
+        private static bool isNoClip = false;
+        public static void SetNoClipState()
         {
             if (PlayerHelper.TryFindPlayers(out List<GameObject> players))
             {
-                foreach(var player in players)
+                foreach (var player in players)
                 {
                     var playerColliders = player.GetComponents<Collider>();
-                    foreach(var collider in playerColliders)
+                    foreach (var collider in playerColliders)
                     {
                         collider.enabled = !isNoClip;
                     }
+                    var playerView = player.GetComponent<PlayerView>();
+                    playerView.Speed = isNoClip ? playerView.Speed = 3000f * SpeedMultiplier.Value : playerView.Speed = 3000f;
```

Comment:
Is 3000 the default speed? Im a little concerned with this value being hardcoded. It would be better to grab this value from the game, then if it ever changes we are not affected. Maybe something like: 

```
public static class NoClipHandler
    {
        private static bool isNoClip = false;
        private static float? defaultPlayerSpeed = null;
        public static void SetNoClipState()
        {
            if (PlayerHelper.TryFindPlayers(out List<GameObject> players))
            {
                foreach (var player in players)
                {
                    var playerColliders = player.GetComponents<Collider>();
                    foreach (var collider in playerColliders)
                    {
                        collider.enabled = !isNoClip;
                    }

                    SetPlayerSpeed(player);
                }
            }
        }

        private static void SetPlayerSpeed(GameObject player)
        {
            var playerView = player.GetComponent<PlayerView>();
            if (defaultPlayerSpeed == null)
            {
                defaultPlayerSpeed = playerView.Speed;
            }
            playerView.Speed = isNoClip ? defaultPlayerSpeed * SpeedMultiplier.Value : defaultPlayerSpeed;
        }

        public static void Update()
        {
            if (NoClipKeyboardShortcut.Value.IsDown())
            {
                isNoClip = !isNoClip;
            }
        }
    }
```

Reply:
Am I supposed to replace the null with 3000f? (the default speed)

Reply:
Naa this part here will just read the existing value (should be 3000) and set it to the defaultPlayerSpeed variable.
 
```
if (defaultPlayerSpeed == null)
{
    defaultPlayerSpeed = playerView.Speed;
}
```
You can use that code exactly as is^^

Reply:
Then when we want to set it back to the starting value we can use that variable :)

Reply:
Ohh yeah just realise we need to cast to float, my bad. Ill push the change. 
```playerView.Speed = (float)(isNoClip ? defaultPlayerSpeed * SpeedMultiplier.Value : defaultPlayerSpeed);```